server/routers/teams.js

```diff
@@ -5,137 +5,122 @@ const Team = context.Team
-        else {
-            try {
-                const team = await Team.create({ name })
-                return res.status(200).send({ "message": "Team " + team.name + " was created." })
-            }
-            catch (err) {
-                return res.status(400).send({ "message": "something bad happened" })
-            }
-        }
+  const name = req.body.name
+  const project_name = req.body.project_name
+
+  //get the username from the session
+  const username = req.username
+
+  //search in the db the user
+  const user = await User.findOne({ where: username })
+  //if the user is a professor, the addition can happen
+  if (user.is_professor === true) {
+    const existantTeam = await Team.findOne({
+      where: {
+        [Op.or]: [name, project_name]
+      }
+    })
+    if (existantTeam !== null) {
+      return res.status(400).send({ "message": "The team name or the project is already taken" })
     }
     else {
-        return res.status(400).send({ "message": apiError.Unauthorized })
+      try {
+        const team = await Team.create({ name, project_name })
+        return res.status(200).send({ "message": "Team " + team.name + " was created." })
+      }
+      catch (err) {
+        return res.status(400).send({ "message": "something bad happened" })
+      }
     }
+  }
+  else {
+    return res.status(400).send({ "message": apiError.Unauthorized })
+  }
 
 });
 
 router.delete('/:id', async (req, res) => {
-    //get the username from the session
-    const username = req.username
-
-    //search in the db the user
-    const user = await User.findOne({ where: username })
-
-    //if the user is a professor, the deletion can happen
-    if (user.is_professor === true) {
-        const id = req.params.id
-        try {
-            const team = await Team.findOne({ where: { id }, include: [ context.User, context.Project ] })
-            team.removeProjects(team.Projects);
-            team.removeUsers(team.Users);
-            await team.destroy()
-            return res.status(200).send({ "message": "Team was deleted" })
-        }
-        catch (err) {
-            return res.status(400).send({ "message": "Something happened at deletion" })
-        }
+  //get the username from the session
+  const username = req.username
+
+  //search in the db the user
+  const user = await User.findOne({ where: username })
+
+  //if the user is a professor, the deletion can happen
+  if (user.is_professor === true) {
+    const id = req.params.id
+    try {
+      const team = await Team.findOne({ where: { id }, include: [context.User, context.Project] })
+      team.removeProjects(team.Projects);
+      team.removeUsers(team.Users);
+      await team.destroy()
+      return res.status(200).send({ "message": "Team was deleted" })
     }
-    else {
-        return res.status(400).send({ "message": apiError.Unauthorized })
+    catch (err) {
+      return res.status(400).send({ "message": "Something happened at deletion" })
```

Comment:
All such responses should be:

```suggestion
      return res.status(500).json(apiError.InternalError);
```